pandas/io/excel/_base.py

```diff
@@ -809,17 +819,24 @@ class ExcelFile:
         "pyxlsb": _PyxlsbReader,
     }
 
-    def __init__(self, io, engine=None):
+    def __init__(self, path_or_io, engine=None):
         if engine is None:
             engine = "xlrd"
+            if isinstance(path_or_io, IOBase):
```

Comment:
I don’t think we need to do this - are we doing this for xls vs .xlsx? Maybe better as a separate PR if you feel strongly and I’m wrong about the existing file types

Reply:
I agree can you revert this

Reply:
No, I do this because of .xls* vs. .ods, and no it cannot be reverted unless tons of tests fail.